### ISSUE TITLE
CMake: wrong library prefix/suffix defaults?

### ISSUE THREAD

**Issue by MarcelHB** [msys2/MINGW-packages#6766]
Yesterday, I downloaded MSYS2 to follow this [building guide of GemRB](https://gemrb.org/MSYS2-build.html) on a plain new Windows 10. Once upon time™, it worked that way.

Now, with the latest MinGW-CMake v.3.17.3 and `-G "MSYS Makefiles"` generation, every call to `find_library()` fails to find the respective library while everything has been installed by pacman into `/mingw64/lib`, like SDL2 or Python 2:
```
libSDL2.dll.a
libSDL2.a
libpython2.7.dll.a
...
```

I then noticed the following CMake variables looking odd:
```
CMAKE_FIND_LIBRARY_PREFIXES=
CMAKE_FIND_LIBRARY_SUFFIXES=.lib
```
That's some kind of Windows convention, but definitely not what the libraries in `/mingw64/lib` are named like. So If I manually insert these things into the project's `CMakeLists.txt`.
```cmake
SET(CMAKE_FIND_LIBRARY_PREFIXES "lib")
SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
```
we're back in business. From my understanding, these should be the defaults on MSYS2? Unfortunately, I can no longer look at my old computer to investigate the last working state there (Windows 7, earlier than July 2020).

**Comment by Biswa96** [comment]
I have tried the [building guide of GemRB](https://gemrb.org/MSYS2-build.html) on a plain new Windows 10 without any error. CMake detects SDL2 and other libraries successfully.

P.S. Suggestion, move to Python 3 and use SDL2 explicitly. Also there are many `--param max-inline-insns-single limit reached` warnings with mingw GCC 10.2.0.

**Comment by MarcelHB** [comment]
Thanks for trying! Yes, your suggestions are already on screen. :wink: 

Anyway, I wonder what else is in play then here?

**Comment by MarcelHB** [comment]
Ok, I think I got something: I noticed that CMake always ended up in `Modules/Platforms/Windows.cmake` loading these wrong defaults, while it's supposed to go to `Modules/Platforms/Windows-GNU.cmake`.

What I missed or need to know:

1. MSYS2 is located on `C:\Program Files\MSYS2`. Writing there requires admin permissions. I previously never had it there.
2. Source code is on `D:\`, so that's good.
3. MSYS2 cannot write to `/tmp`, i. e.  `C:\Program Files\MSYS2\tmp`. without getting proper permissions first, or running MSYS2 as admin.

This changes:
```
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
```
to:
```
```
-- The C compiler identification is GNU 10.1.0
-- The CXX compiler identification is GNU 10.1.0
```

So I guess we can close this.
